src/main/java/fi/vrk/xroad/monitor/actor/MonitorDataActor.java

```diff
@@ -60,32 +66,31 @@ public Receive createReceive() {
                 .build();
     }
 
-    protected void handleMonitorDataRequest(MonitorDataRequest request) {
-        log.info("start handleMonitorDataRequest {}", request.getSecurityServerInfo().toString());
-        String xml = handler.handleMonitorDataRequestAndResponse(request.getSecurityServerInfo());
-        log.info("Response metric: {}", xml);
-        if (xml.startsWith("<m:getSecurityServerMetricsResponse>")) {
-            saveMonitorData(xml, request.getSecurityServerInfo());
+    protected void handleMonitorDataRequest(MonitorDataRequest request)
+        throws ExecutionException, InterruptedException {
```

Comment:
Exceptions are not handler, I think. Maybe theses should have been handled in Supervisor strategy.